moped-toolbox/geometry_prototype/create_prototype.sql

```diff
@@ -0,0 +1,141 @@
+insert into layers (id, internal_table, reference_layer_primary_key_column) values (5, 'intersections', 'intersection_id');
+
+create table component_types (
+    id serial primary key,
+    component_type character varying,
+    subtype character varying,
+    subcomponent character varying,
+    layer_id integer references layers(id) -- think of this layer as the default source layer that users pick geometr from
+    );
+
+-- Massively incomplete list of the valid choices of component_types that can be added. 
+-- Component_type, subtype, and subcomponent are canidates for breaking out into their own tables
+insert into component_types (id, component_type, subtype, subcomponent, layer_id) values (1, 'Signal', 'PHB', null, 3);
+insert into component_types (id, component_type, subtype, subcomponent, layer_id) values (2, 'Signal', 'PHB', 'Audible Push Button', 3);
+insert into component_types (id, component_type, subtype, subcomponent, layer_id) values (3, 'Signal', 'Traffic', null, 3);
+insert into component_types (id, component_type, subtype, subcomponent, layer_id) values (4, 'Sidewalk Install', 'Concrete', null, 4);
+insert into component_types (id, component_type, subtype, subcomponent, layer_id) values (5, 'Intersection Markings', null, null, 5);
+
+create table components (
+    id serial primary key, 
+    project_id integer references projects(id),
+    component_type_id integer references component_types(id),
+    layer_id_override integer references layers(id) default null, -- think of this layer, which is optional, as an override of the default source, and most likly will always be a pointer to 'drawn_points' or 'drawn_lines'
+    name character varying
+    );
+
+-- feature type generic fields related to the geometry could be put here, plus i really like the global ID space it enforces over feature geometry subtables
+create table features (
+    id serial primary key, 
+    name character varying
+    );
+
+-- should we prefix these tables with something to logically group them? schema them?
+create table signals (
+    signal_id integer default null,
+    geography geography('MULTIPOINT') default null
+    ) inherits (features);
+
+create table sidewalks (
+    sidewalk_id integer default null,
+    sidewalk_name character varying default null,
+    geography geography('MULTILINESTRING') default null
+    ) inherits (features);
+
+create table drawn_points (
+    geography geography('MULTIPOINT') default null
+    ) inherits (features);
+
+create table drawn_lines (
+    geography geography('MULTILINESTRING') default null
+    ) inherits (features);
+
+create table component_feature_map (
+    id serial primary key,
+    component_id integer references components(id),
+    feature_id integer default null
+    );
+
+-- this could be (should be?) a CTE to the project_geography view, but this seems easier to read in a prototype
```

Comment:
👍

picking nits, but i think the `table` column here reads better as `layer` or `layer_name` or whatever we end up calling that name column in the `layers` table.